question19/js/slideshow.js

```diff
@@ -0,0 +1,185 @@
+const imgListsWrapper = document.getElementById("js-imgListsWrapper");
+const imgLists = document.getElementById("js-imgLists");
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = "loading";
+  imgListsWrapper.appendChild(loadImg);
+};
+
+const removeLoadImg = () => {
+  document.getElementById("loading").remove();
+};
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch("https://mocki.io/v1/09593e78-80ba-4765-940e-3c70fb46078c"));
+  }, 3000);
+});
+
+//Get json data
+const callApi = async () => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  } catch (error) {
+    console.error(error);
+  } finally {
+    removeLoadImg();
+  }
+};
+
+const init = async () => {
+  const imgData = await callApi();
+  if (imgData) {
+    renderImgUiElement(imgData);
+  } else {
+    imgListsWrapper.textContent = "データの表示に失敗しました";
+    console.error("データの表示に失敗しました");
+  }
+};
+init();
+
+const renderImgUiElement = (imgData) => {
+  createArrowBtnsElements();
+  renderPageNumElement(imgData);
+  renderPagenation(imgData);
+  clickedPagenation(imgData);
+  autoImgSwitch(imgData);
+  arrowBtnsClickEvent(imgData);
+  imgLists.appendChild(getFragmentImglists(imgData));
+};
+
+const renderPageNumElement = (imgData) => {
+  const numberOfPage = createElementWithClassName("p", "pageNumber");
+  imgListsWrapper.appendChild(numberOfPage);
+  numberOfPage.id = "js-number";
+  numberOfPage.textContent = `1 / ${imgData.length}`;
+};
+
+const getFragmentImglists = (imgData) => {
+  const fragmentImglists = document.createDocumentFragment();
+  for (let i = 0; i < imgData.length; i++) {
+    const imgList = createElementWithClassName("li", "imgList");
+    const img = createElementWithClassName("img", `img_0${[i + 1]}`);
+    imgList.dataset.index = i;
+    img.src = imgData[i].img;
+    img.alt = imgData[i].description;
+    i === 0 && imgList.classList.add("is-show");
+    fragmentImglists.appendChild(imgList).appendChild(img);
+  }
+  return fragmentImglists;
+};
+
+const imgNum = { count: 0 };
+const imgList = document.getElementsByClassName("imgList");
+
+const createArrowBtnsElements = () => {
+  const btnDirections = ["next", "prev"];
+  btnDirections.forEach((btnDirections) => {
+    const btn = document.createElement("button");
+    btn.classList.add(`${btnDirections}`, "arrowbotton");
```

Comment:
nits
I'm sorry for being so detailed.
```suggestion
    btn.classList.add(`${btnDirections}`, "arrowbutton");
```

Reply:
Thanks for the review!!! You're right! Fixed it! Please confirm.
fixed commit : https://github.com/csakyo/homework/pull/33/commits/93ece1d5b2158162e5178e82f10838b208e8cc46